tests/unit/utils/test_nova_compute.py

```diff
@@ -66,13 +66,19 @@ async def test_get_instance_count_invalid_result(model, result_key, value):
 @pytest.mark.asyncio
 @patch("cou.utils.nova_compute.get_instance_count")
 async def test_get_empty_hypervisors(
-    mock_instance_count, hypervisors_count, expected_result, model
+    mock_instance_count, hypervisors_count, expected_result, model, caplog
 ):
     mock_instance_count.side_effect = [count for _, count in hypervisors_count]
-    result = await nova_compute.get_empty_hypervisors(
-        [_mock_nova_unit(nova_unit) for nova_unit, _ in hypervisors_count], model
-    )
+    non_empty_log, selected_log = construct_expected_logs(hypervisors_count)
+
+    with caplog.at_level(logging.INFO):
+        result = await nova_compute.get_empty_hypervisors(
+            [_mock_nova_unit(nova_unit) for nova_unit, _ in hypervisors_count], model
+        )
+
     assert {machine.machine_id for machine in result} == expected_result
+    assert non_empty_log in caplog.record_tuples
+    assert selected_log in caplog.record_tuples
```

Comment:
IMO this is overly complex. You could simply mock the logger and check if it has been called with the expected messages. See this test as an example: https://github.com/canonical/charmed-openstack-upgrader/blob/main/tests/unit/test_cli.py#L130-L138